### ISSUE TITLE
RuntimeError: Could not run any SAT solver.

### ISSUE THREAD

**Issue by kalefranz** [conda/conda#8176]
```
error: RuntimeError('Could not run any SAT solver.')
command: C:\Tools\Progs\python\anaconda3\scripts\conda-script.py install -p C:\Tools\Progs\python\anaconda3 -y anaconda-client==1.7.2
user_agent: conda/4.6.1 requests/2.21.0 CPython/3.7.1 Windows/10 Windows/10.0.15063
_messageid: -9223372036617369659
_messagetime: 1548779675918 / 2019-01-29 10:34:35

Traceback (most recent call last):
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\exceptions.py", line 1001, in __call__
    return func(*args, **kwargs)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\cli\main.py", line 84, in _main
    exit_code = do_call(args, p)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\cli\conda_argparse.py", line 81, in do_call
    exit_code = getattr(module, func_name)(args, parser)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\cli\main_install.py", line 21, in execute
    install(args, parser, 'install')
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\cli\install.py", line 250, in install
    force_reinstall=context.force_reinstall or context.force,
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\core\solve.py", line 107, in solve_for_transaction
    force_remove, force_reinstall)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\core\solve.py", line 145, in solve_for_diff
    force_remove)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\core\solve.py", line 240, in solve_final_state
    ssc = self._find_inconsistent_packages(ssc)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\core\solve.py", line 342, in _find_inconsistent_packages
    _, inconsistent_precs = ssc.r.bad_installed(ssc.solution_precs, ())
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\resolve.py", line 896, in bad_installed
    C = r2.gen_clauses()
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\common\io.py", line 85, in decorated
    return f(*args, **kwds)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\resolve.py", line 668, in gen_clauses
    C = Clauses(sat_solver_cls=get_sat_solver_cls(context.sat_solver))
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\_vendor\auxlib\decorators.py", line 56, in _memoized_func
    result = func(*args, **kwargs)
  File "C:\Tools\Progs\python\anaconda3\lib\site-packages\conda\common\logic.py", line 293, in get_sat_solver_cls
    raise RuntimeError('Could not run any SAT solver.')
RuntimeError: Could not run any SAT solver.
```

This shouldn't be a `RuntimeError` that results in an ugly stack trace, but rather some type of `CondaError` that says they've probably managed to uninstall `pycosat`.

**Comment by kalefranz** [comment]
This one is related.  We just need to improve the error handling here.

```
error: RuntimeError('Could not run SAT solver through interface "pycryptosat".')
command: /home/brentp/miniconda3/bin/conda install -y conda-build
user_agent: conda/4.6.2 requests/2.21.0 CPython/3.7.1 Linux/4.15.0-43-generic ubuntu/16.04.5 glibc/2.23
_messageid: -9223372036616970284
_messagetime: 1548885913575 / 2019-01-30 16:05:13

Traceback (most recent call last):
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/common/logic.py", line 279, in get_sat_solver_cls
    cls().run(0)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/common/logic.py", line 175, in run
    solver = self.setup(m, **run_kwargs)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/common/logic.py", line 223, in setup
    from pycryptosat import Solver
ModuleNotFoundError: No module named 'pycryptosat'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/exceptions.py", line 1001, in __call__
    return func(*args, **kwargs)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/cli/main.py", line 84, in _main
    exit_code = do_call(args, p)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/cli/conda_argparse.py", line 82, in do_call
    exit_code = getattr(module, func_name)(args, parser)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/cli/main_install.py", line 20, in execute
    install(args, parser, 'install')
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/cli/install.py", line 253, in install
    force_reinstall=context.force_reinstall or context.force,
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 107, in solve_for_transaction
    force_remove, force_reinstall)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 145, in solve_for_diff
    force_remove)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 240, in solve_final_state
    ssc = self._find_inconsistent_packages(ssc)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/core/solve.py", line 342, in _find_inconsistent_packages
    _, inconsistent_precs = ssc.r.bad_installed(ssc.solution_precs, ())
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 896, in bad_installed
    C = r2.gen_clauses()
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/common/io.py", line 85, in decorated
    return f(*args, **kwds)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/resolve.py", line 668, in gen_clauses
    C = Clauses(sat_solver_cls=get_sat_solver_cls(context.sat_solver))
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/_vendor/auxlib/decorators.py", line 59, in _memoized_func
    result = func(*args, **kwargs)
  File "/home/brentp/miniconda3/lib/python3.7/site-packages/conda/common/logic.py", line 281, in get_sat_solver_cls
    raise RuntimeError('Could not run SAT solver through interface "{}".'.format(name))
RuntimeError: Could not run SAT solver through interface "pycryptosat".
```

**Comment by flutefreak7** [comment]
I got this error today and was befuddled enough to just uninstall and reinstall anaconda. (also some cursing...)

My hunch was that activate wasn't working correctly when doing an env create with an environment.yml file because I also ended up with a mixture of python 3.6 and 3.7 in the base environment (which was on 3.6). When I went to try and fix that base environment, I think this SAT silver error was the next error I got.
